modules/serverless-rn-1-17-0.adoc

```diff
@@ -0,0 +1,53 @@
+[id="serverless-rn-1-17-0_{context}"]
+= Release Notes for Red Hat {ServerlessProductName} 1.17.0
+
+[id="new-features-1-17-0_{context}"]
+== New features
+
+* {ServerlessProductName} now uses Knative Serving 0.23.0.
+* {ServerlessProductName} now uses Knative Eventing 0.23.0.
+* {ServerlessProductName} now uses Kourier 0.23.0.
+* {ServerlessProductName} now uses Knative `kn` CLI 0.23.0.
+* {ServerlessProductName} now uses Knative Kafka 0.23.0.
+* The `kn func` CLI plug-in now uses `func` 0.17.0.
+* In the upcoming {ServerlessProductName} 1.19.0 release, the URL scheme of external routes will default to HTTPS for enhanced security.
++
+If you do not want this change to apply for your workloads, you can override the default setting before upgrading to 1.19.0, by adding the following YAML to your `KnativeServing` custom resource definition (CRD):
++
+[source,yaml]
+----
+...
+spec:
+  config:
+    network:
+      defaultExternalScheme: "http"
+...
+----
+
+* mTLS functionality is now Generally Available (GA).
+
+* TypeScript templates are now available when you create a function using `kn func`.
+
+* Changes to API versions in Knative Eventing 0.23.0:
+
+** Version `v1beta1` has been removed from the `eventing.knative.dev`, `messaging.knative.dev` and `flows.knative.dev` APIs. You must ensure that any existing or new resources in your deployment that are created by using these APIs use version `v1`.
+
+** The `v1alpha1` version of the `KafkaChannel` API, which was deprecated in {ServerlessProductName} version 1.14.0, has been removed. If the `ChannelTemplateSpec` of your config maps contain references to this older version, you must update this part of the spec to use the correct API version.
+
+** The deprecated `v1alpha1`, `v1alpha2`, and `v1beta1` versions of the `APIServerSource` and `SinkBinding` APIs have been removed. You must update any existing resources to use the correct API versions.
+
+** The deprecated `v1alpha2` and `v1beta1` versions of the `ContainerSource` and `PingSource` APIs have been removed. You must update any existing resources to use the correct API versions.
+
+** Version `v1beta2` of the `PingSource` API has been deprecated. References to older versions of this API should be updated to `v1` to prepare your deployment for removal of the deprecated version.
```

Comment:
```suggestion
* Some older versions of eventing APIs are deprecated as show by the following table and the {ServerlessOperatorName} automatically upgrades older resources to the newest versions:

[cols="^.^d, ^d,^d,^d,^d,^d,^d"]
|===
|||v1alpha1|v1alpha2|v1beta1|v1beta2|v1

.4+|sources.knative.dev
|ApiServerSource|deprecated|deprecated|deprecated|-|current
|ContainerSource|-|deprecated|deprecated|-|current
|PingSource|-|deprecated|deprecated|current|served
|SinkBinding|deprecated|deprecated|deprecated|-|current

.3+|messaging.knative.dev
|Channel|-|-|-|-|current
|InMemoryChannel|-|-|-|-|current
|Subscription|-|-|-|-|current

.3+|eventing.knative.dev
|Broker|-|-|-|-|current
|EventType|-|-|-|current|-
|Trigger|-|-|-|-|current

.2+|flows.knative.dev
|Parallel|-|-|-|-|current
|Sequence|-|-|-|-|current
|===

* Deprecated versions can be removed in any upcoming release and you must update your manifests to use the current version.
```

Here's a proposed updated release notes.

Reply:
I like the table, however it should also convey the information about what has changed exactly in Serverless 1.17 ,  esp. the 
```
"Version `v1beta1` has been removed from the `eventing.knative.dev`, `messaging.knative.dev` and `flows.knative.dev`
```
bit. Perhaps by adding "removed" cells in the table?

Reply:
@maschmid done. Check the comment again.

Reply:
This table would be a good addition as a reference module higher up in the doc, i.e. before specific release notes, where we keep it updated each release, but to me as a reader, this doesn't at all tell me what the specific changes were for 1.17.0.

I think we should keep the text as is, and add the table as a different reference module that isn't 1.17.0 specific / can be updated.

Reply:
Also, it's not clear to me what `-` means in the table - does that mean it never existed? It looks odd that some APIs just started life as v1alpha2 or v1beta2. I think in this way the table actually makes things more confusing.

Reply:
@devguyio still awaiting updates on this as discussed in Slack when you have a minute

Reply:
```suggestion
* Changes to Knative Eventing API versions:

** The `v1alpha1` version of the `KafkaChannel` API, which was deprecated in {ServerlessProductName} version 1.14.0, has been removed. If the `ChannelTemplateSpec` of your config maps contain references to this older version, you must update this part of the spec to use the correct API version.

=== Notes
* The {ServerlessOperatorName} automatically upgrades older stored resources from their deprecated version to the newest current version (e.g APIServerSource resources will be upgraded to `v1`) 
* Deprecated versions can be removed in any upcoming release and you must update your manifests to use the current version.
* Trying to use any of the removed versions when creating resources will fail. Make sure to use a current version. Using a deprecated version will not fail but it's not recommended since they will be removed at any future release. 

```

Here's a version that describes the changes to the user that's coming in 1.17.

Reply:
@devguyio I'll add a module to the start of the release notes titled something like "notes on API versions" or something to add these notes to; as per our previous conversations, any general information that applies to all serverless releases should be added somewhere stable as part of the "core" docs, rather than under release notes that only apply for a specific version.
I'll ping you when it's ready for another look.